pin-cushion.js

```diff
@@ -323,8 +486,82 @@ class PinCushion {
             type: Boolean,
             default: false,
             config: true,
+            onChange: s => {
+                // Warn the GM if player notes are allowed while players cannot create Journal Entries
+                if (s === true && game.user.isGM && !game.permissions.JOURNAL_CREATE.includes(1)) {
+                    ui.notifications.warn(game.i18n.format("PinCushion.Warn.AllowPlayerNotes", {permission: game.i18n.localize("PERMISSION.JournalCreate")}));
+                }
+            }
+        });
+
+        game.settings.register(PinCushion.MODULE_NAME, "defaultJournalPermission", {
+            name: "SETTINGS.DefaultJournalPermissionN",
+            hint: "SETTINGS.DefaultJournalPermissionH",
+            scope: "world",
+            type: Number,
+            choices: Object.entries(ENTITY_PERMISSIONS).reduce((acc, [perm, key]) => {
+                acc[key] = game.i18n.localize(`PERMISSION.${perm}`)
+                return acc;
+            }, {}),
+            default: 0,
+            config: true,
             onChange: s => {}
         });
+
+        game.settings.register(PinCushion.MODULE_NAME, "defaultJournalFolder", {
+            name: "SETTINGS.DefaultJournalFolderN",
+            hint: "SETTINGS.DefaultJournalFolderH",
+            scope: "world",
+            type: String,
+            choices: {
+                none: game.i18n.localize("PinCushion.None"),
+                perUser: game.i18n.localize("PinCushion.PerUser"),
+            },
+            default: "none",
+            config: true,
+            onChange: async (s) => {
+                // Only run check for folder creation for the main GM
```

Comment:
I would prefer to break this out into a method on the PinCushion class maybe to avoid overloading the setting change handler block.